src/data-access/apiAccess.ts

```diff
@@ -0,0 +1,163 @@
+import { baseUrl, trendUrl, apiKey, genreListUrl, randomMovieURL, personSearchUrl } from './apiPaths';
+import { MovieDTO } from './movieDTO';
+import { GenreDTO } from './genreDTO';
+import { MovieDetailsDTO } from './movieDetailsDTO';
+import { MovieCastDetailsDTO } from './movieCastDetailsDTO';
+import { MovieTrailersDTO } from './movieTrailersDTO';
+import axiosInstance from './axiosInstance';
+import { RandomMovieDTO } from './randomMovieDTO';
+import { PersonDetailsDTO } from './personDetailsDTO';
+import { PersonPopMoviesDTO } from './personPopMoviesDTO';
+import { SearchPersonDTO } from './searchPersonDTO';
+import { SearchMovieDTO } from './searchMovieDTO';
+
+const API_KEY = `${apiKey}`;
+
+export async function getTrendingMovies(): Promise<MovieDTO[]> {
+    try {
+        const response = await axiosInstance.get(trendUrl);
+        const trendingMovies: MovieDTO[] = response.data.results;
+        return trendingMovies;
+    } catch (error) {
+        console.error('Error fetching trending movies:', error);
+        throw error;
+    }
+}
+
+export async function getGenreList(): Promise<GenreDTO[]> {
+    try {
+        const response = await axiosInstance.get(genreListUrl);
+        const genreList: GenreDTO[] = response.data.genres;
+        return genreList;
+    } catch (error) {
+        console.error('Error fetching genre list:', error);
+        throw error;
+    }
+}
+
+export async function getMovieDetails(show_id: string | undefined): Promise<MovieDetailsDTO> {
+    try {
+        const showUrl = `${baseUrl}movie/${show_id}`;
+        const response = await axiosInstance.get(showUrl, {
+            params: {
+                api_key: API_KEY,
+            },
+        });
+        const movieDetailsResponse: MovieDetailsDTO = response.data;
+        return movieDetailsResponse;
+    } catch (error) {
+        console.error('Error fetching movie details:', error);
+        throw error;
+    }
+}
+
+export async function getMovieCastDetails(show_id: string | undefined): Promise<MovieCastDetailsDTO> {
```

Comment:
Again why we need undefined? you can answer in one of the comments no need to reply each of them